packages/twenty-server/src/database/commands/upgrade-version/0-23/0-23-update-file-folder-structure.command.ts

```diff
@@ -0,0 +1,123 @@
+    private readonly typeORMService: TypeORMService,
+    private readonly dataSourceService: DataSourceService,
+    private readonly workspaceCacheVersionService: WorkspaceCacheVersionService,
+    @InjectDataSource('metadata')
+    private readonly metadataDataSource: DataSource,
+  ) {
+    super();
+  }
+
+  @Option({
+    flags: '-w, --workspace-id [workspace_id]',
+    description: 'workspace id. Command runs on all workspaces if not provided',
+    required: false,
+  })
+  parseWorkspaceId(value: string): string {
+    return value;
+  }
+
+  async run(
+    _passedParam: string[],
+    options: UpdateFileFolderStructureCommandOptions,
+  ): Promise<void> {
+    const workspaceIds = options.workspaceId
+      ? [options.workspaceId]
+      : (await this.workspaceRepository.find()).map(
+          (workspace) => workspace.id,
+        );
+
+    if (!workspaceIds.length) {
+      this.logger.log(chalk.yellow('No workspace found'));
+
+      return;
+    }
+
+    this.logger.log(
+      chalk.green(`Running command on ${workspaceIds.length} workspaces`),
+    );
+
+    for (const workspaceId of workspaceIds) {
+      const dataSourceMetadata =
+        await this.dataSourceService.getLastDataSourceMetadataFromWorkspaceId(
+          workspaceId,
+        );
+
+      if (!dataSourceMetadata) {
+        this.logger.log(
+          `Could not find dataSourceMetadata for workspace ${workspaceId}`,
+        );
+        continue;
+      }
+
+      const workspaceDataSource =
+        await this.typeORMService.connectToDataSource(dataSourceMetadata);
+
+      if (!workspaceDataSource) {
+        throw new Error(
+          `Could not connect to dataSource for workspace ${workspaceId}`,
+        );
+      }
+
+      const workspaceQueryRunner = workspaceDataSource.createQueryRunner();
+
+      const attachmentsToMove = (await workspaceQueryRunner.query(
+        `SELECT id, "fullPath" FROM "${dataSourceMetadata.schema}"."attachment" WHERE "fullPath" LIKE '${FileFolder.Attachment}/%'`,
+      )) as { id: string; fullPath: string }[];
+
+      const workspaceMemberAvatarsToMove = (await workspaceQueryRunner.query(
+        `SELECT id, "avatarUrl" FROM "${dataSourceMetadata.schema}"."workspaceMember" WHERE "avatarUrl" LIKE '${FileFolder.ProfilePicture}/%'`,
+      )) as { id: string; avatarUrl: string }[];
+
+      const personAvatarsToMove = (await workspaceQueryRunner.query(
+        `SELECT id, "avatarUrl" FROM "${dataSourceMetadata.schema}"."person" WHERE "avatarUrl" LIKE '${FileFolder.PersonPicture}/%'`,
+      )) as { id: string; avatarUrl: string }[];
+
+      const workspacePictureToMove = await this.workspaceRepository.findOneBy({
+        id: workspaceId,
+        logo: Like(`${FileFolder.WorkspaceLogo}/%`),
+      });
+
+      this.logger.log({
+        workspacePictureToMove,
+        attachmentsToMove,
+        workspaceMemberAvatarsToMove,
+        personAvatarsToMove,
+      });
+
+      // TODO: move files and update paths
```

Comment:
I hope this "v0.23" approach of migration command is correct.

Now what I dont know how to do is how to actually move the files. We only have `write` and `read` methods on the interface. Shall I add one more like "move" and implement it in this PR as well? Or just "delete" and do "read" -> "write" -> "delete"? WDYT @charlesBochet?

Reply:
That's a good point. I think implementing move in the driver makes sense if it's not too complicated
Thanks for the amazing work